src/lib/insights/client.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  Hits,
+  InsightsClient,
+  InsightsClientMethod,
+  InsightsClientPayload,
+  InsightsClientWrapper,
+  Renderer,
+  RenderOptions,
+  Unmounter,
+  WidgetFactory,
+  Without,
+} from '../../types';
+import find from 'lodash/find';
+import uniq from 'lodash/uniq';
+
+const getSelectedHits = (hits: Hits<unknown>, selectedObjectIDs: string[]) => {
+  return selectedObjectIDs.map(objectID => {
+    const hit = find(hits, h => h.objectID === objectID);
+    if (typeof hit === 'undefined') {
+      throw new Error(
+        `Could not find ObjectID ${objectID} passed to 'clickedObjectIDsAfterSearch' in the returned hits. This is necessary to infer absolute position and queryID.`
+      );
+    }
+    return hit;
+  });
+};
+
+const getQueryID = (selectedHits: Hits<unknown>) => {
+  const queryIDs = uniq(selectedHits.map(hit => hit.__queryID));
```

Comment:
It's not the first time we use `uniq` but for a simple case like this one `indexOf` work fine.